cypress/integration/download.cy.js

```diff
@@ -0,0 +1,74 @@
+import {
+    DIMENSION_ID_ENROLLMENT_DATE,
+    DIMENSION_ID_EVENT_DATE,
+} from '../../src/modules/dimensionConstants.js'
+import {
+    HIV_PROGRAM,
+    ANALYTICS_PROGRAM,
+    TEST_REL_PE_LAST_12_MONTHS,
+} from '../data/index.js'
+import {
+    selectEnrollmentProgram,
+    selectEventProgram,
+} from '../helpers/dimensions.js'
+import { clickMenubarUpdateButton } from '../helpers/menubar.js'
+import { selectRelativePeriod } from '../helpers/period.js'
+import { EXTENDED_TIMEOUT } from '../support/util.js'
+
+const downloadIsEnabled = () =>
+    cy
+        .getBySel('menubar')
+        .contains('Download')
+        .should('not.have.attr', 'disabled')
+
+const downloadIsDisabled = () =>
+    cy.getBySel('menubar').contains('Download').should('have.attr', 'disabled')
+
+describe('download', () => {
+    it('download button enables when required dimensions are selected (event)', () => {
+        cy.visit('/', EXTENDED_TIMEOUT)
+
+        downloadIsDisabled()
+
+        selectEventProgram({
+            programName: HIV_PROGRAM.programName,
+            stageName: HIV_PROGRAM.stageName,
+        })
+
+        clickMenubarUpdateButton()
+
+        downloadIsDisabled()
+
+        selectRelativePeriod({
+            label: HIV_PROGRAM[DIMENSION_ID_EVENT_DATE],
+            period: TEST_REL_PE_LAST_12_MONTHS,
+        })
+
+        clickMenubarUpdateButton()
```

Comment:
I would add a `expectTableToBeVisible()` between the update and the download being enabled, as I assume that's what ultimately determines if you can download the table, right? (e.g. if for some reason the steps taken above isn't enough to generate a table, then the button should be disabled still and the test should fail)

Reply:
I think it's okay to enable the download button as soon as the required steps for passing the layout validation are met. In the app, whether analytics returned data or not e.g. for the selected period is kind of out of our hands. We could potentially look at `store.loader.loadingError` in combination with `store.current`, but not sure if we want to go down that road.